libexec/bugfiler/sendbug.1
.\"	BSDI sendbug.1,v 2.3 1996/01/19 22:54:50 bostic Exp
.\"
.\" Copyright (c) 1983, 1990, 1993
.\"	The Regents of the University of California.  All rights reserved.
.\"
.\" Redistribution and use in source and binary forms, with or without
.\" modification, are permitted provided that the following conditions
.\" are met:
.\" 1. Redistributions of source code must retain the above copyright
.\"    notice, this list of conditions and the following disclaimer.
.\" 2. Redistributions in binary form must reproduce the above copyright
.\"    notice, this list of conditions and the following disclaimer in the
.\"    documentation and/or other materials provided with the distribution.
.\" 3. All advertising materials mentioning features or use of this software
.\"    must display the following acknowledgement:
.\"	This product includes software developed by the University of
.\"	California, Berkeley and its contributors.
.\" 4. Neither the name of the University nor the names of its contributors
.\"    may be used to endorse or promote products derived from this software
.\"    without specific prior written permission.
.\"
.\" THIS SOFTWARE IS PROVIDED BY THE REGENTS AND CONTRIBUTORS ``AS IS'' AND
.\" ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
.\" IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
.\" ARE DISCLAIMED.  IN NO EVENT SHALL THE REGENTS OR CONTRIBUTORS BE LIABLE
.\" FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
.\" DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
.\" OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
.\" HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
.\" LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
.\" OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
.\" SUCH DAMAGE.
.\"
.\"     @(#)sendbug.1	8.1 (Berkeley) 6/4/93
.\"
.Dd June 4, 1993
.Dt SENDBUG 1
.Os BSD 4.2
.Sh NAME
.Nm sendbug
.Nd mail a system bug report to problem@BSDI.COM
.Sh SYNOPSIS
.Nm sendbug
.Op Ar address
.Sh DESCRIPTION
Bug reports sent to
.Dq problem@BSDI.COM
are intercepted by a program which expects bug reports to conform to
a standard format.
The
.Nm sendbug
command is a shell script to help users compose and mail bug reports
in the correct format.
.Pp
.Nm Sendbug
works by invoking the editor specified by the environment variable
.Ev EDITOR
on a temporary copy of the bug report format outline.
The user must fill in the appropriate fields.
If the file has been changed when the user exits the editor,
.Nm sendbug
then mails the completed report to the
.Ar address
specified on the command line, or to
.Dq problem@BSDI.COM
by default.
.Sh ENVIRONMENT
The following environment variables affect the execution of
.Nm sendbug :
.Bl -tag -width BUGADDR
.It Ev BUGADDR
The address to which the bug report should be sent
when no arguments are given.
If
.Ev BUGADDR
is not set,
.Nm sendbug
defaults to
.Li problem@BSDI.COM .
.It Ev EDITOR
Specifies the preferred editor.
If
.Ev EDITOR
is not set,
.Nm sendbug
defaults to
.Xr vi 1 .
.El
.Sh FILES
.Bl -tag -width /usr/share/misc/bugformat -compact
.It Pa /usr/share/misc/bugformat
Contains the bug report outline.
.El
.Sh SEE ALSO
.Xr vi 1 ,
.Xr bugfiler 8
.Sh HISTORY
The
.Nm sendbug
command
appeared in
.Bx 4.2 .


libexec/bugfiler/sendbug.sh
#!/bin/sh -
#
#	BSDI sendbug.sh,v 2.4 2000/02/29 00:22:25 polk Exp
#
# Copyright (c) 1983, 1993
#	The Regents of the University of California.  All rights reserved.
#
# Redistribution and use in source and binary forms, with or without
# modification, are permitted provided that the following conditions
# are met:
# 1. Redistributions of source code must retain the above copyright
#    notice, this list of conditions and the following disclaimer.
# 2. Redistributions in binary form must reproduce the above copyright
#    notice, this list of conditions and the following disclaimer in the
#    documentation and/or other materials provided with the distribution.
# 3. All advertising materials mentioning features or use of this software
#    must display the following acknowledgement:
#	This product includes software developed by the University of
#	California, Berkeley and its contributors.
# 4. Neither the name of the University nor the names of its contributors
#    may be used to endorse or promote products derived from this software
#    without specific prior written permission.
#
# THIS SOFTWARE IS PROVIDED BY THE REGENTS AND CONTRIBUTORS ``AS IS'' AND
# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
# ARE DISCLAIMED.  IN NO EVENT SHALL THE REGENTS OR CONTRIBUTORS BE LIABLE
# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
# OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
# HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
# OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
# SUCH DAMAGE.
#
#	@(#)sendbug.sh	8.1 (Berkeley) 6/4/93
#

# create a bug report and mail it to 'problem@bsdi.com'.

# internet sites
DEFADDR=problem@BSDI.COM
# uucp sites need a suitable path here
#DEFADDR=bsdi.com!problem

PATH=/bin:/sbin:/usr/sbin:/usr/bin:/usr/contrib/bin:/usr/local/bin:$PATH
export PATH

BUGTEMP=/tmp/bug$$
FORMATFILE=/usr/share/misc/bugformat

trap 'rm -f $BUGTEMP $BUGTEMP.0 ; exit 1' 1 2 3 13 15
trap 'rm -f $BUGTEMP $BUGTEMP.0 ; exit 0' 0

case $# in
0) set ${BUGADDR-${DEFADDR}} ;;
esac

(echo To: "$@"; echo Cc: ${USER}; cat $FORMATFILE) > $BUGTEMP
cp ${BUGTEMP} ${BUGTEMP}.0
chmod u+w $BUGTEMP
if ${EDITOR=vi} $BUGTEMP
then
	if cmp -s ${BUGTEMP}.0 $BUGTEMP
	then
		echo "File not changed, no bug report submitted."
		exit
	fi
	sendmail -t -oi < $BUGTEMP
fi
